### PR TITLE
Add a volume-weighted Weibull IC for randomized local strength

### LINKED ISSUE #10221: Add volume-weighted Weibull initial condition
## Rationale
Add a volume-weighted Weibull distribution initial condition to describe material strength following the description in this paper:

O. Strack, R. Leavy, and R. Brannon. Aleatory uncertainty and scale effects in computational damage models for failure and fragmentation. International Journal for Numerical Methods in Engineering, 102(3-4):468–495, Apr. 2015.
http://doi.wiley.com/10.1002/nme.4699

## Description
Using this distribution to define the material strength permits simulations to realistically replicate fracture localization and can alleviate size dependencies of fracture simulations. It is a standard Weibull distribution, but includes a dependency on element size, so this does not naturally fit in the standard RandomIC.

## Impact
I propose that we simply add a new IC to tensor_mechanics, since that is the place where this will be used. This can exist independent of the regular RandomIC, which we would like to expand to work with various distributions.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #10221

### CODE REVIEW

File: modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  [u_vww]
+    order = CONSTANT
+    family = MONOMIAL
+  []
+[]
+
+[ICs]
+  [u_vww]
+    type = VolumeWeightedWeibull
+    legacy_generator = false
```
Comment (jiangwen84) [inline modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i]: we should probably remove this. The default is false and this option will be deprecated.

File: modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    outputs = initial
+  [../]
+[]
+
+[Executioner]
+  type = Steady
+[]
+
+[Outputs]
+  exodus = true
```
Comment (jiangwen84) [inline modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i]: I thought you used only CSVDiff. Do we need output exodus here?

File: modules/tensor_mechanics/src/ics/VolumeWeightedWeibull.C [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VolumeWeightedWeibull.h"
+
+#include "libmesh/point.h"
+#include "Distribution.h"
```
Comment (jiangwen84) [inline modules/tensor_mechanics/src/ics/VolumeWeightedWeibull.C]: Did you use Distribution here?

**Review by jiangwen84 — COMMENTED** [review]
Looks great. Minor comments.

File: modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i (hunk 1 above)
Comment (bwspenc) [reply modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i]: Removed it.

File: modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i (hunk 2 above)
Comment (bwspenc) [reply modules/tensor_mechanics/test/tests/ics/volume_weighted_weibull/volume_weighted_weibull.i]: Agreed -- removed it.

File: modules/tensor_mechanics/src/ics/VolumeWeightedWeibull.C (hunk 3 above)
Comment (bwspenc) [reply modules/tensor_mechanics/src/ics/VolumeWeightedWeibull.C]: No.  Removed it.

**Comment by bwspenc** [comment]
@jiangwen84 Thanks for the review.  I addressed all of your issues.

@aeslaughter If you have a chance, would you mind reviewing this and merging if it looks good to you? We discussed this a while back. Because of the volume correction that this does, it really needs to be done in its own class rather than within RandomIC. Since this is intended to be used for distributions of material strength, and is used for fracture modeling, I think it makes sense to put it in tensor_mechanics, but I'm open to putting it somewhere else if you think that would be better.

**Review by permcody — APPROVED** [review]

**Pull request merged**